design/workflow.md

```diff
@@ -54,14 +54,14 @@ jobs:
 
 ### SCM Branch-specific jobs
 
-Use a regex filter after `~commit:` to denote a job only run when code changes are made to branches matching that pattern (eg: `/^staging$/`, `/^user-.*$/`).
+Use exact match of a string (eg: `staging`) or a slash-quoted regex filter (eg: `/user-.*/`) after `~commit:` to denote a job only run when code changes are made to branches matching that pattern　(eg: `staging`, `/^user-.*$/`).
```

Comment:
Yes, does `requires: ~commit:/user-.*/` translate into `requires: ~commit:/^user-.*$/` in Screwdriver.
Also now I think it will be better to allow user to write `requires: ~commit:/^user-.*$/`.

So user can use `requires: ~commit:/user-.*/` and `requires: ~commit:/^user-.*$/` and both  translate into  `requires: ~commit:/^user-.*$/` in Screwdriver.

How do you think about this idea?

Reply:
If the idea is looks good, I will fix the sentence like 
```
... or a slash-quoted regex filter (eg: `/user-.*/` or `/^user-.*$/`) 
```

Reply:
I think it's kind of confusing. What if the user wants to use regex exactly `/user-.*/`? Then it should match branches where it doesn't start with `user-` as well.

I think we should just have these cases:
1. Implied exact match: `requires: ~commit:staging` => `requires: ~commit:/^staging$/` 
2. Literal regex: If there's // at the beginning and end, we should take the regex directly (and not do any extra parsing).
`requires: ~commit:/user-.*/` => `requires: ~commit:/user-.*/`
`requires: ~commit:/^user-.*$/` => `requires: ~commit:/^user-.*$/`

Reply:
It will be better to use regex exactly.
But the exact match is better not to use regex in Screwdriver, because branch able to have characters of `.`, `+` and so on. 

I think the first case would be
1. Implied exact match:  `requires: ~commit:staging` => `requires: ~commit:staging` and compare by the same way as it used to

I agree with the second case.

Reply:
Sounds good